### ISSUE TITLE
Move DCP/1 to a different subdomain to make way for the release of DCP/2

### ISSUE THREAD

**Issue by theathorn** [DataBiosphere/data-portal#717]
@NoopDog @hannes-ucsc Here's the plan for releasing DCP/2 to the public:
  - Continue development on DCP/2 using dev.singlecell.gi.ucsc.edu and prod.singlecell.<etc>
  - Add an alias for DCP/1 (data.<etc>) at v1.data.humancellatlas.org
  - On the DCP/2 home page add a "Click here for old version of the DCP" which links to v1.data.<etc>
  - On the day of DCP/2 release, redirect data.humancellatlas.org to the prod.singlecell.<etc> instance
  - Some months later retire DCP/1 by removing the link from the DCP/2 home page and retiring the "v1.data" sub-domain

Rationale:
  - It biases people to go over to the new data, too, which is what we want.
  - Old project URLs would still work but the user will get the DCP/2 version of the project. But that seems compatible with the DCP/1 mode of "you always see the latest version - we don't have snapshots or data releases yet".
  - We don't need to get a new sub-domain of humancellatlas.org assigned from the Sanger
  - We don't need to get the Sanger to update the link to the DCP that's on humancellatlas.org

I'm sure we need ticket(s) to standup the Azul service in DCP/1 under "v1.data.<etc>".

**Comment by theathorn** [comment]
No, the user's address bar should read "data.humancellatlas.org". So, "reassign" the URL?

When we do this, will we break other DCP/1 services and endpoints such as https://matrix.data.humancellatlas.org/?

We need to have a good plan..

**Comment by NoopDog** [comment]
One way this could be done is to:

1. For the portal, just update the content to the new content and not try to show any older versions of content and leave that on data.humancellatlas.org

1. For the browser leave this also on data.humancellatlas.org/explore but allow the user to select which back end they want to use. By default we would use the /v2 api but could allow the user to select to use the /v1 api.

Then the browser would call the correct back end service depending on if the user selected v1 vs v2. We could make it very clear when in V1 mode that this is the old version of the data. 

Azul could deploy on the same prod prod endpoint and use a /v1 or /v2 API version to determine which back end really gets the message. This could be setup with cloudfront or  likely API Gateway. 

On the browser we could also update the NEW urls to indicate that they require the new backend. like 

https://data.humancellatlas.org/explore/v2/projects/f86f1ab4-1fbb-4510-ae35-3ffd752d4dfc


For old links like
https://data.humancellatlas.org/explore/projects/f86f1ab4-1fbb-4510-ae35-3ffd752d4dfc
that exist on the internet we would want to catch those and redirect to the new version but indicate that the old version exists and provide a link to it.

Keeping both versions on the same URL likely gives us the least pain and most flexibility in how to handle things and keeps things simple for the user and existing URLs linking back to us. 

Whatever we decide  needs a proper write up in a google doc but wanted to get out a sketch of a proposal in this ticket. 
Basically this proposal is don't use a new v1.data sub domain. Instead, use a version number in the path.

**Comment by theathorn** [comment]
We want "data." to resolve to the DCP/2 instance without any added "2" etc. in the URL.

Users with bookmarked/cited URLs will resolve to e.g. the latest instance of a project in DCP/2.

DCP/1 Portal/Browser/Azul be moved to the sub-domain "v1.data." which will only exist temporarily during the overlap period with DCP/2. 

"matrix.data." and "dss.data." can resolve to the DCP/1 service instances ("how?" is TBD).

The AWS Route 53 name servers for data.humancellatlas.org are in the CZI HCA account (hca-id).
We have admin access to the account but CZI would need to maintain/pay for the account to maintain the name resolution for "data." Should be very low $$. Alternatively we would have to go back to Sanger to have them tell the Registrar (Mesh Digital at domainbox.com) to point to name servers under one of our AWS accounts (could be UCSC, Broad or EBI). So either CZI needs to keep the hca-id account in perpetuity or we need to go back to Sanger to transfer "data.humancellatlas.org" to name servers that we control.

**Comment by theathorn** [comment]
MattW (CZI) says:
  - Our strong preference is to close all the old AWS accounts
  - looks like we’re paying ~$110 / month in route53 costs

**Comment by theathorn** [comment]
See also DataBiosphere/azul#2014.

**Comment by theathorn** [comment]
@NoopDog @hannes-ucsc Can we configure the DCP/1 domain to be v1.data.humancellatlas.org now and (for the short term) redirect data.humancellatlas.org to v1.data.humancellatlas.org? i.e. Users going to "data.<>" will see "v1.data.<>' in the address bar.

Then when DCP/2 goes live, we make "data.humancellatlas.org" the one-and-only URL for DCP/2 and "v1.data.humancellatlas.org" as the one-and-only URL for DCP/1.

Then at the end of the year, we tell CZI they can tear down the CZI HCA accounts (and v1.data.humancellatlas.org along with it).

**Comment by hannes-ucsc** [comment]
I think that should be possible with a 301 redirect in the origin S3 bucket that backs the DB's CloudFront distro.

https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html

@NoopDog owns that bucket.

**Comment by theathorn** [comment]
After talking with Kathleen today, we're no longer going to setup a "v1.data" sub-domain since the DCP/1 will be retired by CZI at the end of December. Instead we'll send an emails notifying the general HCA community and the portals community of the impending shutdown. When DCP/2 is ready in December we'll switch over the "data.humancellatlas.org" sub-domain to the Data Browser and Azul instances in the UCSC AWS account.

**Comment by MillenniumFalconMechanic** [comment]
Closing as per standup.
